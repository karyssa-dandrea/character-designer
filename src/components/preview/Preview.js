import React from 'react';

export default function Preview({
  head,
  middle,
  bottom,
  phrase,
  headCount,
  middleCount,
  bottomCount,
  phraseCount,
}) {
  return (
    <div className={`preview ${phrase}`}>
      <p>{head}</p>
      <p>{middle}</p>
      <p>{bottom}</p>
      <p>{phrase}</p>
      <div>
        <p>You have changed the head {headCount} times</p>
        <p> You have changed the middle {middleCount} times</p>
        <p>You have changed the bottom {bottomCount} times</p>
        <p>You have added a catch phrase {phraseCount} times </p>
      </div>
    </div>
  );
}
