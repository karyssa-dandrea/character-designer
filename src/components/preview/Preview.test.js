import { render, screens } from '@testing-library/react';
import Preview from './Preview';

test('renders the preview panel', () => {
  const container = render(
    <Preview
      head=""
      middle=""
      bottom=""
      phrase=""
      headCount="0"
      middleCount="0"
      bottomCount="0"
      phraseCount="0"
      phraseList={[]}
    />
  );
  expect(container).toMatchSnapshot();
});
