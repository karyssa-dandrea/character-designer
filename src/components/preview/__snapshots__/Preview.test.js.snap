// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the preview panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="preview "
      >
        <img
          alt=""
          height="150"
          src="/outfits/.png"
        />
        <img
          alt=""
          height="150"
          src="/outfits/-middle.png"
        />
        <img
          alt=""
          height="150"
          src="/outfits/.png"
        />
        <div
          id="phrase-2"
        />
        <div>
          <p>
            You have changed the head 
            0
             times
          </p>
          <p>
             You have changed the middle 
            0
             times
          </p>
          <p>
            You have changed the bottom 
            0
             times
          </p>
          <p>
            You have added a catch phrase 
            0
             times 
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="preview "
    >
      <img
        alt=""
        height="150"
        src="/outfits/.png"
      />
      <img
        alt=""
        height="150"
        src="/outfits/-middle.png"
      />
      <img
        alt=""
        height="150"
        src="/outfits/.png"
      />
      <div
        id="phrase-2"
      />
      <div>
        <p>
          You have changed the head 
          0
           times
        </p>
        <p>
           You have changed the middle 
          0
           times
        </p>
        <p>
          You have changed the bottom 
          0
           times
        </p>
        <p>
          You have added a catch phrase 
          0
           times 
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
